test/functional/rpc_net.py

```diff
@@ -168,12 +172,96 @@ def test_getnettotals(self):
             self.wait_until(lambda: peer_after()['bytesrecv_per_msg'].get('pong', 0) >= peer_before['bytesrecv_per_msg'].get('pong', 0) + 32, timeout=1)
             self.wait_until(lambda: peer_after()['bytessent_per_msg'].get('ping', 0) >= peer_before['bytessent_per_msg'].get('ping', 0) + 32, timeout=1)
 
+    def test_getnetmsgstats(self):
+        self.log.info("Test getnetmsgstats")
+
+        # Compare byte count with what getnettotals returns
+        getnettotals_response = self.nodes[0].getnettotals()
+        getnetmsgstats_response = self.nodes[0].getnetmsgstats()
+
+        assert_equal(getnettotals_response['totalbytessent'], getnetmsgstats_response['sent']['total']['total_bytes'])
+        assert_equal(getnettotals_response['totalbytesrecv'], getnetmsgstats_response['received']['total']['total_bytes'])
+
+        # Test filter aggregation with a few different combinations
+        getnetmsgstats_msgtype = self.nodes[0].getnetmsgstats(filters=["msgtype"])
+        assert_array_result([getnetmsgstats_msgtype['sent']['ping']], {'msg_count': 4}, {'total_bytes': 128})
+        assert_array_result([getnetmsgstats_msgtype['received']['pong']], {'msg_count': 4}, {'total_bytes': 128})
+
+        getnetmsgstats_conntype_msgtype = self.nodes[0].getnetmsgstats(filters=["conntype", "msgtype"])
+        assert_array_result([getnetmsgstats_conntype_msgtype['received']['manual']['pong']], {'msg_count': 2}, {'total_bytes': 64})
+
+        getnetmsgstats_conntype_msgtype_network = self.nodes[0].getnetmsgstats(filters=["conntype", "msgtype", "network"])
+        manualconn_sendcmpct_stats = [{'msg_count': 2}, {'total_bytes': 66}]
+        assert_array_result([getnetmsgstats_conntype_msgtype_network['sent']['manual']['sendcmpct']['not_publicly_routable']],
```

Comment:
Is it okay to test against the "not_publicly_routable" network type?

Also, this line fails for me after a fresh build, but never afterwards. Should I wait and see what the CI says?

Reply:
Failing tests are fixed by disconnecting from all peers at the beginning of the test. See https://github.com/vasild/bitcoin/commit/8c6920ae89e6dc031a278211854fb7518436976c